tests/test_utils/test_schemaview.py

```diff
@@ -1036,40 +1036,67 @@ def test_enums_and_enum_relationships(schema_view_no_imports: SchemaView) -> Non
+    assert view.permissible_value_ancestors("CAT", animals) == ["CAT"]
+    assert set(view.permissible_value_children("CAT", animals)) == {"LION", "TABBY"}
+    assert set(view.permissible_value_descendants("CAT", animals)) == {"CAT", "LION", "ANGRY_LION", "TABBY"}
+
+    pv_tabby = animal_enum.permissible_values["TABBY"]
+    assert pv_tabby.is_a == "CAT"
+    assert view.permissible_value_parent("TABBY", animals) == ["CAT"]
+    assert view.permissible_value_ancestors("TABBY", animals) == ["TABBY", "CAT"]
+    assert view.permissible_value_children("TABBY", animals) == []
+    assert view.permissible_value_descendants("TABBY", animals) == ["TABBY"]
+
+    pv_lion = animal_enum.permissible_values["LION"]
+    assert pv_lion.is_a == "CAT"
+    assert view.permissible_value_parent("LION", animals) == ["CAT"]
+    assert view.permissible_value_ancestors("LION", animals) == ["LION", "CAT"]
+    assert view.permissible_value_children("LION", animals) == ["ANGRY_LION"]
+    assert view.permissible_value_descendants("LION", animals) == ["LION", "ANGRY_LION"]
+
+    pv_angry_lion = animal_enum.permissible_values["ANGRY_LION"]
+    assert pv_angry_lion.is_a == "LION"
+    assert view.permissible_value_parent("ANGRY_LION", animals) == ["LION"]
+    assert view.permissible_value_ancestors("ANGRY_LION", animals) == ["ANGRY_LION", "LION", "CAT"]
+    assert view.permissible_value_children("ANGRY_LION", animals) == []
+    assert view.permissible_value_descendants("ANGRY_LION", animals) == ["ANGRY_LION"]
+
+
+# FIXME: improve testing of dynamic enums
+def test_dynamic_enum(schema_view_with_imports: SchemaView) -> None:
+    """Rudimentary test of dynamic enum."""
+    view = schema_view_with_imports
 
-    for cn, c in view.all_classes().items():
-        if c.name == "Adult":
-            assert view.class_ancestors(c.name) == ["Adult", "Person", "HasAliases", "Thing"]
+    # dynamic enums
+    e = view.get_enum("HCAExample")
+    assert set(e.include[0].reachable_from.source_nodes) == {"GO:0007049", "GO:0022403"}
 
 
-def test_schemaview(schema_view_no_imports: SchemaView) -> None:
-    """General SchemaView tests."""
+def test_get_elements_applicable_by_identifier(schema_view_no_imports: SchemaView) -> None:
+    """Test get_elements_applicable_by_identifier method."""
     view = schema_view_no_imports
-    logger.debug(view.imports_closure())
-    assert len(view.imports_closure()) == 1
+    elements = view.get_elements_applicable_by_identifier("ORCID:1234")
+    assert PERSON in elements
+    elements = view.get_elements_applicable_by_identifier("PMID:1234")
+    assert "Organization" in elements
+    elements = view.get_elements_applicable_by_identifier("http://www.ncbi.nlm.nih.gov/pubmed/1234")
+    assert "Organization" in elements
+    elements = view.get_elements_applicable_by_identifier("TEST:1234")
+    assert "anatomical entity" not in elements
 
-    all_cls = view.all_classes()
-    logger.debug(f"n_cls = {len(all_cls)}")
```

Comment:
least favourite test anti-pattern: printing out stuff and not doing any testing on it.